packages/web-components/fast-foundation/src/data-grid/data-grid-row.ts

```diff
@@ -262,7 +310,23 @@ export class DataGridRow extends FoundationElement {
                     e.preventDefault();
                 }
                 break;
+
+            case keySpace:
+                e.preventDefault();
+                this.toggleSelected(e.shiftKey ? true : !this.selected, e);
+                break;
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public handleClick(e: MouseEvent): void {
+        if (e.defaultPrevented || !this.clickSelect) {
+            return;
         }
+        e.preventDefault();
+        this.toggleSelected(e.shiftKey ? true : !this.selected, e);
```

Comment:
I was wondering how you might pre-select items. I was able to add aria-selected="true" to a row. But the click behavior is weird because you're tracking this.selected rather than this.getAttribute("aria-selected"). It presumes that it's starting in a non-selected state. It might be more robust to forego using this.selected and just check the aria-selected.

Reply:
I want to manage selection at the parent grid level as that can work for virtualization as well.  I'll setup pre-selection.

Reply:
Authors can now specify initial selection via an attribute:
`element.setAttribute("initial-selection", "1, 2")`

and can get/set selection via a `selectedRowIndexes` property:
`(element as DataGrid).selectedRowIndexes = [1];`